### ISSUE TITLE
Allow enforcing camelCase

### ISSUE THREAD

**Issue by xevix** [daniel-trinh/scalariform#98]
Periodically developers who come and go between snake_case languages and Scala mistakenly use snake_case. A preference to enforce this would be great.

From the code I've seen camelCase is the norm in Scala, and it's also featured in the style guide http://docs.scala-lang.org/style/naming-conventions.html.

I'd be willing to help with this if it makes sense. Thanks.

**Comment by jawshooah** [comment]
How would you handle snake_case identifiers from external libraries that we have no power to change?

What you're asking for is actual static analysis, which is not the purpose of scalariform. Try scalastyle instead; it has rules for naming conventions.

**Comment by xevix** [comment]
You're right, I wasn't targeting external libraries, only the codebase I'm immediately working on. Typically I've used sbt-scalariform which runs Scalariform on each compile.

Interesting, I didn't know this would require more than Scalariform already does for e.g. `alignParameters` but I may have been somewhat naive. Can scalastyle reformat code to match style, or does it only print warnings? I'm more interested in the ability to automatically switch snake_case to camelCase, than in reporting.

**Comment by jawshooah** [comment]
Scalastyle is just a linter, it doesn't reformat anything.

`alignParameters` doesn't need to understand the semantics of your program, it operates purely on syntactic elements. To do what you want, scalariform would need to ensure that a snake_case identifier is defined in your own code base rather than an external library. It would also need to make sure that the new identifier doesn't conflict with any others in the namespace.

Basically, you're looking for an IDE feature :wink:

**Comment by xevix** [comment]
Ah, then yes, it seems like neither would really do what I want. An IDE feature would be nice, but I like the enforcement at compile time that Scalariform can do with formatting. I wonder if a Scala compiler plugin would do the trick. A bit of a hardcore route, but might be interesting.

Meanwhile, closing this issue. Thanks!

**Comment by jawshooah** [comment]
No problem. And I just realized a better way to summarize the problem: scalariform is a *reformatting* tool. What you want is a *refactoring* tool.

**Comment by xevix** [comment]
Yeah, that's a great way to think about it. I naively thought that since we're talking about the "format" of a variable name that it should be simple, but I wasn't even considering all the scope-related problems. Thanks for the informative answers!
